.github/workflows/header-gen.yml

```diff
@@ -18,15 +29,30 @@ jobs:
       uses: actions/setup-node@v4
       with:
         node-version: '20'
+        registry-url: "https://npm.pkg.github.com"
+        cache: npm
+        cache-dependency-path: header-gen/package-lock.json
 
     - name: Install Dependencies
       run: npm ci
       working-directory: ./header-gen
 
+    - name: Update version
+      if: github.event_name != 'pull_request'
```

Comment:
This step will also run on any pushes with changes to the header-gen source code, even if they are not meant to update the package version yet (which likely has already increased as we `pxt bump` the extension), is that the intention?

I assume this action only creates a commit if the version in package.json and the version in the last repo tag don't match?

Reply:
In this case, the action should create a published version with something like @0.1.0-[branch_name].[build_number]. It should never create any commits.

Reply:
But looking at the description in https://github.com/microbit-foundation/npm-package-versioner-action it sounds like if the package.json version entry needs to be updated it will do that? Does it do it without committing the change to the repo?

So, it changes package.json, builds package, pushes it to the package registry with a temporary tag like `@0.1.0-[branch_name].[build_number]` and doesn't commit anything? At what point it will push a package with the correct version number?

Reply:
sorry, the publishing is done by the `npm publish` step.

So, the microbit-foundation/npm-package-versioner-action action will change the package.json version entry if needed (maybe to `0.1.0-[branch_name].[build_number]` as you mentioned), without committing anything and then it's up to other steps to create and/or publish the package. Is that right?

How does it decide if the version entry in package.json needs to be changed? is it if this commit is tagged AND the tag version is the same as the version in package.json?

Reply:
Apologies. This is ignorance on my part. We've not used this in prod yet and I naively assumed it would work in the same way as the package it's trying to replace.

Reply:
Great, thanks Rob!